### DISCUSSION TITLE
Model used in `emotion` downstream

### DISCUSSION THREAD

**Discussion by bagustris** [s3prl/s3prl#426] [General]
Hi @leo19941227 

I would ask/clarify the model used for `emotion` downstream. In this [issue](https://github.com/s3prl/s3prl/issues/302), you mentioned that `er` task only used linear layers with average pooling. Checking these lines, we'll see that these configs (linear layer + pooling) are shown in line 37-41.

https://github.com/s3prl/s3prl/blob/8b63c5a764382c8979a61c853f76c84e10c81492/s3prl/downstream/emotion/config.yaml#L36-L49

My questions are:
1. Is this true: `er` used a fully connected network/linear layer with a single layer and 256 units/nodes (two layers if the output layer with 4 nodes is counted)?
2. What is the role of `DeepModel` and its configuration (`model_type`, etc) in the config above? Is it used (or not used?) when I run `er` downstream using the default config?

**Comment by leo19941227** [comment]
Hi @bagustris ,

1. Yes your are correct.
2. The DeepModel was used to boost the performance on IEMOCAP when using fbank feature. It is an available model for emotion downstream however is is not used for the config you paste. You need to change the `select` field to `DeepModel` to use the DeepModel

**Comment by celsofranssa** [comment]
Hello,
Could you give me the directions to correct define the `config.yaml` to use models like [HuBERT](https://huggingface.co/facebook/hubert-large-ls960-ft)?

**Comment by bagustris** [comment]
@celsofranssa 

The upstream (this is what you meant by models like HuBERT) is not defined in `config.yaml`, but when running the program, i.e., 

```
python3 run_downstream.py -m train -n iemocap-hubert-large -u hubert_large_ll60k -d emotion
```

The argument `-u` define what upstream is used, you can name the experiment with `-n` with any name you want.

**Reply by celsofranssa** [reply]
great, thank you.
